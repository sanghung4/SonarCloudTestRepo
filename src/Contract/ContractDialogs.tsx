import { useContext, useMemo } from 'react';

import { noop } from 'lodash-es';
import { useTranslation } from 'react-i18next';

import { ContractContext, DialogEnum } from 'Contract/ContractProvider';

import CommonAlert from 'common/Alerts';
import { useCartContext } from 'providers/CartProvider';

type Props = {
  count: number;
};

type DialogConfigMemo = {
  [key: string]: {
    title: string;
    message: string;
    confirmBtnTitle?: string;
    onConfirm?: () => void;
    onCancel?: () => void;
    cancelBtnTitle?: string;
  };
};

export default function ContractDialogs(props: Props) {
  /**
   * Custom Hooks
   */
  const { t } = useTranslation();

  /**
   * Context
   */
  const {
    contractData,
    dialogType: dialogueType = '',
    goToCart,
    resetDialog,
    setQtyInputMap,
    qtyInputMap
  } = useContext(ContractContext);
  const { releaseContractToCart } = useCartContext();

  /**
   * Memo
   */
  const { title, message, confirmBtnTitle, onConfirm, cancelBtnTitle } =
    useMemo(configMemo, [
      dialogueType,
      props.count,
      setQtyInputMap,
      t,
      qtyInputMap,
      contractData,
      goToCart,
      releaseContractToCart
    ]);

  /**
   * Render
   */
  return (
    <CommonAlert
      title={title}
      message={message}
      confirmBtnTitle={confirmBtnTitle}
      cancelBtnTitle={cancelBtnTitle}
      onConfirm={handleConfirm}
      onCancel={resetDialog}
      open={!!dialogueType}
    />
  );

  /*
  Callbacks
  */
  function handleConfirm() {
    onConfirm?.();
    resetDialog();
  }

  /**
   * Memo Def
   */
  function configMemo() {
    const configs = {
      [DialogEnum.Clear]: {
        title: t('contracts.clearQtyTitle'),
        message: t('contracts.clearQtyDesc', {
          qty: Object.keys(qtyInputMap)?.length
        }),
        confirmBtnTitle: t('contracts.clearConfirm'),
        cancelBtnTitle: t('contracts.backToContract'),
        onConfirm: () => setQtyInputMap({})
      },
      [DialogEnum.Release]: {
        title: t('contracts.releaseAllTitle'),
        message: t('contracts.releaseAllDesc', { qty: props.count }),
        confirmBtnTitle: t(`contracts.continueRelease`),
        onConfirm: () => {
          releaseContractToCart(contractData);
          goToCart();
        },
        cancelBtnTitle: t('contracts.backToContract')
      },
      [DialogEnum.SubtotalToLarge]: {
        title: t('contracts.releaseAllOver10MilTitle'),
        message: t('contracts.releaseAllOver10MilDesc'),
        cancelBtnTitle: t('contracts.backToContract')
      },
      '': {
        title: '',
        message: '',
        confirmBtnTitle: '',
        onConfirm: noop
      }
    } as DialogConfigMemo;
    return configs[dialogueType];
  }
}
