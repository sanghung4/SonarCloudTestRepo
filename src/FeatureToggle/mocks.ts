import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
