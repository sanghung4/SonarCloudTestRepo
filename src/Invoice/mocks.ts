import { GetInvoiceDocument } from 'generated/graphql';

export const successMincron = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '123456',
      invoiceNumber: '5460026'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5460026',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: '40251/13325',
        invoiceDate: '10/28/2020',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'AVLNEP7',
        jobName: 'NEP 7 ASHEVILLE',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2020',
        deliveryMethod: 'Our Truck',
        subtotal: '3184.28',
        tax: '262.76',
        otherCharges: '0.0',
        paidToDate: '1000.00',
        invoiceItems: [
          {
            id: '1738433F4-91A7-4637-8C60-B4DF59A060C1',
            brand: 'Advanced Drainage Systems, Inc.',
            name: '120 x 7.5, 100 Sq Yd, Geosynthetic, Double, Photodegradable, Erosion Control Blanket',
            partNumber: '8X1125STRAWDN',
            mfr: '00S2TT',
            thumb: 'null',
            price: '43.59',
            qty: {
              quantityOrdered: 1,
              quantityShipped: 1
            }
          }
        ]
      }
    }
  }
};

export const successMincron_5448540 = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '91492',
      invoiceNumber: '5448540'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5282411',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: 'WILKESBORO JIM',
        invoiceDate: '10/06/2021',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'SHOP',
        jobName: 'SHOP',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2021',
        deliveryMethod: 'Our Truck',
        subtotal: '2184.28',
        tax: '162.73',
        otherCharges: '0.0',
        paidToDate: '139.91',
        invoiceItems: {
          id: 'F02FD85F-4DCE-4EA7-8459-2DA7CFD689DC',
          brand: 'Certex USA, name:2 x 12, Nylon, Eye and Eye, Ply Sling',
          partNumber: 'SLING212,mfr:CX07-0126-12',
          thumb:
            'https://images.tradeservice.com/ProductImages/DIR100193/CERTEX_CX07-0126-12_LINE_SML.jpg',
          price: '40.00',
          qty: {
            quantityOrdered: 1,
            quantityShipped: 1
          }
        }
      }
    }
  }
};
