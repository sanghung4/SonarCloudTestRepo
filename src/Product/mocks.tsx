import {
  GetProductDocument,
  GetProductPricingDocument
} from 'generated/graphql';

export const success = {
  request: {
    query: GetProductDocument,
    variables: {
      productInput: { productId: 'MSC-33860', customerNumber: undefined }
    }
  },
  result: {
    data: {
      product: {
        id: 'MSC-33860',
        partNumber: '33860',
        name: '1/2" Wrot Copper Cap',
        manufacturerName: 'Generic',
        manufacturerNumber: 'CCAPD',
        price: 0,
        stock: null,
        taxonomy: null,
        categories: [
          'Pipe & Fittings',
          'Metal Pipe & Fittings',
          'Copper Tube & Fittings'
        ],
        technicalDocuments: [
          {
            name: 'MFR Catalog Page',
            url: 'http://images.tradeservice.com/ATTACHMENTS/DIR100145/MUELERE00056_18_20.pdf',
            __typename: 'TechDoc'
          },
          {
            name: 'Technical Specification',
            url: 'http://images.tradeservice.com/ATTACHMENTS/DIR100240/MUELERE03088_1.pdf',
            __typename: 'TechDoc'
          },
          {
            name: 'MFR Item Data',
            url: 'http://images.tradeservice.com/ATTACHMENTS/DIR100145/MUELERE00601_1.pdf',
            __typename: 'TechDoc'
          }
        ],
        environmentalOptions: [],
        upc: '68576823241',
        unspsc: '40183104',
        seriesModelFigureNumber: null,
        productOverview:
          'Cap Fitting; Type Round Head; Nominal Size 1/2"; End Connection Female Soldered; Material Copper; Process Wrot; Application Ground Potable Water Supply System; Applicable Standard ASME B16.22, ANSI/NSF 61 G; Packaging Quantity 50 per Inner Pack, 1000 per Master Carton',
        featuresAndBenefits:
          'Copper Solder-Joint Fittings for Supply/Pressurized Systems have Been the Leading Brand of Copper Fittings for Over 80 Years; Available in Both Wrot Copper and Cast Bronze; The Acknowledged Experts at Engineering and Manufacturing Precision Solder-Joint Copper Fittings; Quality, Consistency and Reliability have Made the Streamline Brand Trusted and Specified All Around the World',
        techSpecifications: [
          {
            name: 'material',
            value: 'Copper',
            __typename: 'TechSpec'
          },
          {
            name: 'Applicable Standard',
            value: 'ASME B16.22, ANSI/NSF 61 G',
            __typename: 'TechSpec'
          },
          {
            name: 'Application',
            value: 'Ground Potable Water Supply System',
            __typename: 'TechSpec'
          },
          {
            name: 'End Connection',
            value: 'Female Soldered',
            __typename: 'TechSpec'
          },
          {
            name: 'Material',
            value: 'Copper',
            __typename: 'TechSpec'
          },
          {
            name: 'Nominal Size',
            value: '1/2"',
            __typename: 'TechSpec'
          },
          {
            name: 'Process',
            value: 'Wrot',
            __typename: 'TechSpec'
          },
          {
            name: 'Type',
            value: 'Round Head',
            __typename: 'TechSpec'
          }
        ],
        imageUrls: {
          thumb:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_SML.jpg',
          small:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_SML.jpg',
          medium:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_MED.jpg',
          large:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_LRG.jpg',
          __typename: 'ImageUrls'
        },
        packageDimensions: null,
        minIncrementQty: 0,
        erp: 'ECLIPSE',
        __typename: 'Product'
      }
    }
  }
};

export const pricing = {
  request: {
    query: GetProductPricingDocument,
    variables: {
      input: {
        customerId: '',
        branchId: 'any',
        productIds: ['33860'],
        includeListData: false
      }
    }
  },
  result: {
    data: {
      productPricing: {
        customerId: 'testAccount',
        branch: 'any',
        products: [
          {
            productId: '33860',
            orderUom: 'ea',
            catalogId: '',
            branchAvailableQty: 2,
            totalAvailableQty: 3,
            listIds: [],
            sellPrice: 3.256
          }
        ]
      }
    }
  }
};
