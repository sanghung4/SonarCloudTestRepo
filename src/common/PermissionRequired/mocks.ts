import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
