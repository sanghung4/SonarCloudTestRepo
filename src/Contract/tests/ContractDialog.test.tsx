import { act, fireEvent } from '@testing-library/react';

import { mockCartContext } from 'Cart/tests/mocks';
import ContractDialogs from 'Contract/ContractDialogs';
import {
  ContractContext,
  ContractContextType,
  DialogEnum
} from 'Contract/ContractProvider';
import { ContractsProviderMocks } from 'Contract/tests/mocks';
import { CartContext, CartContextType } from 'providers/CartProvider';
import { setBreakpoint } from 'test-utils/mockMediaQuery';
import { render } from 'test-utils/TestWrapper';

const mockProvider: ContractContextType = { ...ContractsProviderMocks };
const mockCart: CartContextType = { ...mockCartContext };

describe('Contract - Dialog', () => {
  it('Matches snapshot as "Release All" in desktop', () => {
    setBreakpoint('desktop');
    mockProvider.dialogType = DialogEnum.Release;
    const wrapper = render(
      <ContractContext.Provider value={mockProvider}>
        <ContractDialogs count={1} />
      </ContractContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Matches snapshot as "Release All" in mobile', () => {
    setBreakpoint('mobile');
    mockProvider.dialogType = DialogEnum.Release;
    const wrapper = render(
      <ContractContext.Provider value={mockProvider}>
        <ContractDialogs count={1} />
      </ContractContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Matches snapshot as "undefined"', () => {
    setBreakpoint('desktop');
    mockProvider.dialogType = undefined;
    const wrapper = render(
      <ContractContext.Provider value={mockProvider}>
        <ContractDialogs count={1} />
      </ContractContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Matches snapshot as "Clear All"', () => {
    setBreakpoint('desktop');
    mockProvider.dialogType = DialogEnum.Clear;
    const wrapper = render(
      <ContractContext.Provider value={mockProvider}>
        <ContractDialogs count={1} />
      </ContractContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Confirm button action as "Release All"', async () => {
    mockProvider.dialogType = DialogEnum.Release;
    const wrapper = render(
      <ContractContext.Provider value={mockProvider}>
        <ContractDialogs count={1} />
      </ContractContext.Provider>,
      { cartConfig: mockCart }
    );

    const button = wrapper.getByTestId('alert-confirm-button');
    fireEvent.click(button);
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(mockProvider.resetDialog).toBeCalledTimes(1);
    expect(mockProvider.goToCart).toBeCalledTimes(1);
    expect(mockCart.releaseContractToCart).toBeCalledTimes(1);
    expect(mockCart.releaseContractToCart).toBeCalledWith(
      mockProvider.contractData
    );
  });

  it('Confirm button action as "Release All"(Fortiline and value over 10 million)', async () => {
    mockProvider.dialogType = DialogEnum.SubtotalToLarge;
    const wrapper = render(
      <ContractContext.Provider value={mockProvider}>
        <ContractDialogs count={1} />
      </ContractContext.Provider>,
      { cartConfig: mockCart }
    );

    const button = wrapper.getByTestId('alert-cancel-button');
    fireEvent.click(button);
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(mockProvider.resetDialog).toBeCalledTimes(1);
  });

  it('Confirm button action as "Clear All"', async () => {
    mockProvider.dialogType = DialogEnum.Clear;
    const wrapper = render(
      <ContractContext.Provider value={mockProvider}>
        <ContractDialogs count={1} />
      </ContractContext.Provider>
    );

    const button = wrapper.getByTestId('alert-confirm-button');
    fireEvent.click(button);
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(mockProvider.setQtyInputMap).toBeCalledTimes(1);
  });
});
