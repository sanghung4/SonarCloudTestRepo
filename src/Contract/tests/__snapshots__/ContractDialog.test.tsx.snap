// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contract - Dialog Matches snapshot as "Clear All" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1200px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiModal-root MuiDialog-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-15p4qre-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby="mui-4"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-6gkcvy-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <div
            class="MuiBox-root css-gg4vpm"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-1s0mlu3-MuiTypography-root"
            >
              Clear All Quantities
            </h5>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-1unv2py-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg>
                delete.svg
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1x06mp7-MuiTypography-root"
            >
              You are about to clear the quantities on all the items in your contract that have a quantity for release (0). Are you sure you want to continue with clearing?
            </p>
            <div
              class="MuiBox-root css-n1yaga"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-ku8rdk-MuiButtonBase-root-MuiButton-root"
                data-testid="alert-cancel-button"
                tabindex="0"
                type="button"
              >
                Back To Contract
              </button>
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1egkuzh-MuiButtonBase-root-MuiButton-root"
                data-testid="alert-confirm-button"
                tabindex="0"
                type="button"
              >
                Yes, Clear All Qtys
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Contract - Dialog Matches snapshot as "Release All" in desktop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1200px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiModal-root MuiDialog-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-15p4qre-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby="mui-1"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-6gkcvy-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <div
            class="MuiBox-root css-gg4vpm"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-1s0mlu3-MuiTypography-root"
            >
              Release All Items
            </h5>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-1unv2py-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg>
                delete.svg
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1x06mp7-MuiTypography-root"
            >
              You are about to release all the available items (1) in your contract. Are you sure you want to continue with releasing?
            </p>
            <div
              class="MuiBox-root css-n1yaga"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-ku8rdk-MuiButtonBase-root-MuiButton-root"
                data-testid="alert-cancel-button"
                tabindex="0"
                type="button"
              >
                Back To Contract
              </button>
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1egkuzh-MuiButtonBase-root-MuiButton-root"
                data-testid="alert-confirm-button"
                tabindex="0"
                type="button"
              >
                Continue to Release
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Contract - Dialog Matches snapshot as "Release All" in mobile 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 414px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiModal-root MuiDialog-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-15p4qre-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby="mui-2"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-6gkcvy-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <div
            class="MuiBox-root css-gg4vpm"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-1s0mlu3-MuiTypography-root"
            >
              Release All Items
            </h5>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-1unv2py-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg>
                delete.svg
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1x06mp7-MuiTypography-root"
            >
              You are about to release all the available items (1) in your contract. Are you sure you want to continue with releasing?
            </p>
            <div
              class="MuiBox-root css-14fd8zk"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-ku8rdk-MuiButtonBase-root-MuiButton-root"
                data-testid="alert-cancel-button"
                tabindex="0"
                type="button"
              >
                Back To Contract
              </button>
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1egkuzh-MuiButtonBase-root-MuiButton-root"
                data-testid="alert-confirm-button"
                tabindex="0"
                type="button"
              >
                Continue to Release
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Contract - Dialog Matches snapshot as "undefined" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style=""
  >
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
